### PR TITLE
fix(megamenu): remove menu role for accessibility

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Web component: Mega Menu - "The essentials, Hybrid Cloud" are read as menu options for screen reader users. #5081

### Description

This PR removes the `role='menu'` from the `ul` as it is unnecessary and conflicts with the elements placed in the megamenu/dropdown.

PR here from carbon-for-ibm-dotcom where we removed the menu role from our own version of the header menu
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/4624

The dropdown menu/megamenu shouldn't have the role="menu" applied as listed here:
Example Disclosure for Navigation Menus | WAI-ARIA Authoring Practices 1.1
https://www.w3.org/TR/wai-aria-practices/examples/disclosure/disclosure-navigation.html

Screen Shot 2020-12-03 at 4 25 20 PM

### Changelog

**Changed**

- grab the `bx--header-menu` element and remove the `role` attribute

### CODE REVIEW

File: packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts [hunk 1]
```diff
@@ -76,6 +82,7 @@ class DDSMegaMenuTopNavMenu extends DDSTopNavMenu {
 
   updated(changedProperties) {
     super.updated(changedProperties);
+    this._menuNode.removeAttribute('role');
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:85]: Would you do this in `firstUpdated()` given it doesn't have to do with a property update?

**Review by asudoh — COMMENTED** [review]
Thank you for isolating the change @annawen1!

File: packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts (hunk 1 above)
Comment (annawen1) [reply packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:85]: ah yes! will do

**Review by asudoh — APPROVED** [review]
Thanks @annawen1 for the update!

**Pull request merged**